WordPress/src/main/java/org/wordpress/android/ui/JetpackConnectionDeeplinkActivity.java

```diff
@@ -91,48 +88,35 @@ protected void onCreate(Bundle savedInstanceState) {
     }
 
     @Override
-    protected void onStop() {
-        mDispatcher.unregister(this);
-        super.onStop();
+    protected void onActivityResult(int requestCode, int resultCode, Intent data) {
+        super.onActivityResult(requestCode, resultCode, data);
+        if (requestCode == RequestCodes.JETPACK_LOGIN && resultCode == RESULT_OK) {
+            trackResult();
+        }
+        finishAndGoBackToSource();
     }
 
     @Override
-    protected void onStart() {
-        super.onStart();
-        mDispatcher.register(this);
+    public void onBackPressed() {
+        super.onBackPressed();
+        finishAndGoBackToSource();
```

Comment:
In this PR's current form, this will tell `STATS` that user is logged in with Jetpack which in most cases won't be true.

Note: We might make a change to that flow, which will be mentioned in a separate comment, in which case my comment might not apply.

Reply:
I think it will be true in most cases. The deeplink will only be triggered if the user is logged in to Jetpack. I haven't reached a state yet where the deeplink was called and the user wasn't logged in (on Calypso) with Jetpack. The problem is the app is not yet synced.

Reply:
> I think it will be true in most cases.